client/components/button-group/test/index.js

```diff
@@ -1,35 +1,19 @@
 /**
  * External dependencies
  */
-import { assert } from 'chai';
 import { shallow } from 'enzyme';
 import React from 'react';
-import sinon from 'sinon';
 
 /**
  * Internal dependencies
  */
-
 import { Button } from '@automattic/components';
+import ButtonGroup from '..';
 
 describe( 'ButtonGroup', () => {
-	let sandbox, ButtonGroup;
-
-	beforeEach( () => {
-		sandbox = sinon.createSandbox();
-		sandbox.stub( console, 'error' );
-		sandbox.stub( console, 'log' );
-
-		ButtonGroup = require( '../index' );
-	} );
-
-	afterEach( () => {
-		sandbox.restore();
-	} );
-
 	test( 'should have ButtonGroup class', () => {
 		const buttonGroup = shallow( <ButtonGroup /> );
-		assert.equal( 1, buttonGroup.find( '.button-group' ).length );
+		expect( buttonGroup.find( '.button-group' ) ).toHaveLength( 1 );
```

Comment:
For future reference: the `jest-enzyme` package contains [many convenience matchers](https://www.npmjs.com/package/jest-enzyme#tocontainexactlyonematchingelement) for Enzyme testers.

Reply:
Thanks!

I tried

```diff
diff --git a/client/components/button-group/test/index.js b/client/components/button-group/test/index.js
index 447f47a77c..a9f5426193 100644
--- a/client/components/button-group/test/index.js
+++ b/client/components/button-group/test/index.js
@@ -13,7 +13,7 @@ import ButtonGroup from '..';
 describe( 'ButtonGroup', () => {
        test( 'should have ButtonGroup class', () => {
                const buttonGroup = shallow( <ButtonGroup /> );
-               expect( buttonGroup.find( '.button-group' ) ).toHaveLength( 1 );
+               expect( buttonGroup ).toContainExactlyOneMatchingElement( '.button-group' );
        } );
 
        test( 'should contains the same number of .button nodes than <Button>s it receives', () => {
```

but that gave me `TypeError: expect(...).toContainExactlyOneMatchingElement is not a function`. I guess we need to set up `jest-enzyme` (maybe in `calypso-build`)?

Reply:
In the `calypso-build` Jest preset, `jest-enzyme` is already set up. But the `test/client` Jest setup doesn't use the preset 🙁. `jest-enzyme` needs to be added there separately to make it work.

Also, `test/client` doesn't load the `jest-enzyme` module directly, but rather uses a trick with mocking the `enzyme` module and setting up related modules lazily, only when a test really uses Enzyme. I don't know how much faster (if even) are the test runs with this optimization.

Reply:
Ah, right. I forgot that some of our test setup is still a bit exotic.

Since this seems to be a bit of a :rabbit: :hole: , I'll merge this PR as-is for now :grimacing: